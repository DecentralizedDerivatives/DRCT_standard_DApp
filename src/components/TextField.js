import React from 'react';
import PropTypes from 'prop-types';

const TextField = ({
  id,
  type,
  className,
  placeholder,
  label,
  name,
  value,
  onChange,
  disabled,
  helperText
}) => {
  return (
    <div className="input-field">
      <input
        id={id}
        type={type}
        className={className}
        placeholder={placeholder}
        name={name}
        value={value}
        onChange={onChange}
        disabled={disabled}
      />
      <label for={id}>{label}</label>
      <span class="helper-text" data-error="wrong" data-success="right">
        {helperText}
      </span>
    </div>
  );
};

TextField.propTypes = {
  id: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  className: PropTypes.string,
  placeholder: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
  onChange: PropTypes.func,
  disabled: PropTypes.string,
  helperText: PropTypes.string
};

TextField.defaultProps = {
  type: 'text'
};

export default TextField;
