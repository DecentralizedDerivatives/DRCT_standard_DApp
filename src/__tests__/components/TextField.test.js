// COMPLETE
import TextField from '../../components/TextField';

function setup(overrides) {
  const store = initStore();

  const id = 'testId';
  const name = 'testName';
  const type = 'text';
  const meta = { touched: true, error: 'required' };

  const props = { store, id, name, type, meta, ...overrides };

  const wrapper = shallow(<TextField {...props} />);

  return {
    wrapper,
  };
}

describe('<TextField />', () => {
  it('renders the component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
