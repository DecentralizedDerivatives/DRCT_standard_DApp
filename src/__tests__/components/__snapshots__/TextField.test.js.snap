// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextField
    id="testId"
    name="testId"
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    type="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          id="testId"
          name="testId"
          type="text"
        />,
        <label
          for="testId"
        />,
        <span
          class="helper-text"
          data-error="wrong"
          data-success="right"
        />,
      ],
      "className": "input-field",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": undefined,
          "disabled": undefined,
          "id": "testId",
          "name": "testId",
          "onChange": undefined,
          "placeholder": undefined,
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "for": "testId",
        },
        "ref": null,
        "rendered": null,
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "class": "helper-text",
          "data-error": "wrong",
          "data-success": "right",
        },
        "ref": null,
        "rendered": null,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            id="testId"
            name="testId"
            type="text"
          />,
          <label
            for="testId"
          />,
          <span
            class="helper-text"
            data-error="wrong"
            data-success="right"
          />,
        ],
        "className": "input-field",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": undefined,
            "disabled": undefined,
            "id": "testId",
            "name": "testId",
            "onChange": undefined,
            "placeholder": undefined,
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "for": "testId",
          },
          "ref": null,
          "rendered": null,
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "class": "helper-text",
            "data-error": "wrong",
            "data-success": "right",
          },
          "ref": null,
          "rendered": null,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
